test/MockContract.js

```diff
@@ -31,30 +43,254 @@ contract('MockContract', function(accounts) {
+  });
+
+  describe("givenOutOfGasAny", function() {
+    it("should run out of gas if mocked", async function() {
+      const mock = await MockContract.new();
+      const complex = ComplexInterface.at(mock.address)
+
+      const methodId = "0x" + abi.methodID("acceptAdressUintReturnBool", ["address", "uint"]).toString("hex")
+      await mock.givenOutOfGasAny(methodId);
+
+      // On error it should return the error message for a call
+      var encoded = await complex.contract.acceptAdressUintReturnBool.getData("0x0", 10);
+      error = await utils.getErrorMessage(complex.address, 0, encoded)
+      assert.equal(error, "")
+      encoded = await complex.contract.acceptAdressUintReturnBool.getData("0x1", 12);
+      error = await utils.getErrorMessage(complex.address, 0, encoded)
+      assert.equal(error, "")
+
+      await utils.assertOutOfGas(complex.acceptAdressUintReturnBool("0x0", 10))
+      await utils.assertOutOfGas(complex.acceptAdressUintReturnBool("0x1", 12))
+
+      // Check that we can reset revert
+      await mock.reset()
+      // Transactions should be successful
+      await complex.acceptAdressUintReturnBool("0x0", 10)
+      await complex.acceptAdressUintReturnBool("0x1", 12)
+    });
+  });
+
+  describe("test mock priority", function() {
+
+    const methodId = "0x" + abi.methodID("acceptUintReturnString", ["uint"]).toString("hex")
+    const testSpecificMocks = async function (mock, complex) {
+      const encoded = await complex.contract.acceptUintReturnString.getData(42)
+      await mock.givenReturn(encoded, abi.rawEncode(['string'], ["return specific"]).toString());
+      result = await complex.acceptUintReturnString.call(42);
+      // Specific mock should be prioritized over any mock
+      assert.equal(result, "return specific")
+
+      await mock.givenRevert(encoded);
+      await utils.assertRevert(complex.acceptUintReturnString(42))
+
+      await mock.givenRevertWithMessage(encoded, "revert specific");
+      error = await utils.getErrorMessage(complex.address, 0, encoded)
+      assert.equal(error, "revert specific")
+
+      await mock.givenOutOfGas(encoded);
+      await utils.assertOutOfGas(complex.acceptUintReturnString(42))
+
+      // Check that we can reset revert
+      await mock.reset()
+      // Transactions should be successful
+      result = await complex.acceptUintReturnString.call(42);
+      assert.equal(result, "")
+    }
+
+    it("all specific mocks should be prioritized over return any mock", async function() {
+      const mock = await MockContract.new();
+      const complex = ComplexInterface.at(mock.address)
+
+      // No mock set
+      result = await complex.acceptUintReturnString.call(42);
+      assert.equal(result, "")
+
+      await mock.givenReturnAny(methodId, abi.rawEncode(['string'], ["return any"]).toString());
+      result = await complex.acceptUintReturnString.call(42);
+      assert.equal(result, "return any")
+
+      await testSpecificMocks(mock, complex)
```

Comment:
I'm generally not a huge fan of factoring test cases to share common steps/asserts, since it makes understanding the test case for future maintainer harder. 
I think it's easiest if a test reads as one story (setup, interaction, asserts), without having to jump around in the code into helper functions.
But I'm fine leaving it in this case, as it's very scoped to the describe block.